mantidproject/mantid
# Flips D22 and D33 detector pixel numbering

**Description of work.**

The pixel numbering was once again wrong for the above instruments.
This PR fixes it in the IDFs, also adjusts the loader for the erroneously swapped data for left and right panels of D33. The mask files and the system test references are updated.
The corresponding change in the IDF generators is in PR [#160](https://github.com/mantidproject/mantidgeometry/pull/160) in geometry.



**To test:**
1. Clear `~/.mantid/instrument` folder.
2. LoadEmptyInstrument with D22, D22lr and D33 IDFs in `/instrument'
3. Open the instrument viewer for each, make sure that the default view is 3D seen from sample (i.e. Z-)
4. Open the pick tab and mouse hover over the pixels.
5. Make sure that for each panel (D22 has 1, D33 has 5) the detector IDs start from bottom left corner, and grow first upwards, then to the right.

Note that for D33 picking will be slow, you have to be patient. This is being handled in a separate PR.



Fixes #27727 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
